packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -273,6 +361,7 @@ export class Cluster extends Resource implements ICluster {
 
     this.addAutoScalingGroup(asg, {
       maxPods: maxPodsForInstanceType(options.instanceType),
+      mapRole: options.mapRole,
     });
 
     return asg;
```

Comment:
If a user wants to add IAM statements to the role for the worker nodes they would use this return right?

```ts
const iamEnabledNodes = eksCluster.addCapacity( ... )
iamEnabledNodes.addToRolePolicy( // special IAM permissions for managing IAM to pods // )
```

I've seen a lot of patterns where teams run sensitive workloads on a specific node group, that they might have previously run on master nodes.

Reply:
I think this can be resolved. The ergonomics here are a little awkward for me. Add hands back an entity that you can't access from the class. However, I think the use case is met.